Header header
int32 do_ch1
int32 do_ch2
int32 do_ch3
int32 do_ch4


